cloudferrylib/os/actions/check_cloud.py

```diff
@@ -163,9 +165,17 @@ def run(self, **kwargs):
             LOG.error(err_message)
             raise exception.AbortMigrationError(err_message)
         compute_resource.nova_client.servers.delete(vm_new_ids.keys()[0])
+        for i in xrange(LIMIT_RETRY_CHECK_INSTANCE):
+            if compute_resource.nova_client.servers.list(vm_new_ids.keys()[0]):
```

Comment:
It will return list of ALL vms. Please use `search_opts={'id': vm_new_ids.keys()[0]}`.